src/installer.ts

```diff
@@ -65,41 +114,63 @@ export async function getNode(versionSpec: string) {
   core.addPath(toolPath);
 }
 
-async function queryLatestMatch(versionSpec: string): Promise<string> {
-  // node offers a json list of versions
-  let dataFileName: string;
-  switch (osPlat) {
-    case 'linux':
-      dataFileName = `linux-${osArch}`;
-      break;
-    case 'darwin':
-      dataFileName = `osx-${osArch}-tar`;
-      break;
-    case 'win32':
-      dataFileName = `win-${osArch}-exe`;
-      break;
-    default:
-      throw new Error(`Unexpected OS '${osPlat}'`);
+async function getInfoFromManifest(
+  versionSpec: string,
+  stable: boolean,
+  token: string
+): Promise<INodeVersionInfo | null> {
+  let info: INodeVersionInfo | null = null;
+  const releases = await tc.getManifestFromRepo(
+    'actions',
+    'node-versions',
+    token,
+    'update-versions-manifest-file' // TODO: remove after testing
+  );
+  console.log(`matching ${versionSpec}...`);
+  const rel = await tc.findFromManifest(versionSpec, stable, releases);
+
+  if (rel && rel.files.length > 0) {
+    info = <INodeVersionInfo>{};
+    info.resolvedVersion = rel.version;
+    info.downloadUrl = rel.files[0].download_url;
+    info.fileName = rel.files[0].filename;
+    info.token = token;
   }
 
-  let versions: string[] = [];
-  let dataUrl = 'https://nodejs.org/dist/index.json';
-  let httpClient = new hc.HttpClient('setup-node', [], {
-    allowRetries: true,
-    maxRetries: 3
-  });
-  let response = await httpClient.getJson<INodeVersion[]>(dataUrl);
-  let nodeVersions = response.result || [];
-  nodeVersions.forEach((nodeVersion: INodeVersion) => {
-    // ensure this version supports your os and platform
-    if (nodeVersion.files.indexOf(dataFileName) >= 0) {
-      versions.push(nodeVersion.version);
-    }
-  });
+  return info;
+}
 
-  // get the latest version that matches the version spec
-  let version: string = evaluateVersions(versions, versionSpec);
-  return version;
+async function getInfoFromDist(
+  versionSpec: string
+): Promise<INodeVersionInfo | null> {
+  let osPlat: string = os.platform();
+  let osArch: string = translateArchToDistUrl(os.arch());
+
+  let info: INodeVersionInfo | null = null;
+  let version: string;
+
+  version = await queryDistForMatch(versionSpec);
+  if (!version) {
+    return null;
+  }
+
+  //
```

Comment:
This is all mostly the same code - just moved into another function for fallback.